x-pack/platform/plugins/shared/lens/public/app_plugin/lens_document_equality.ts

```diff
@@ -45,8 +45,10 @@ export const isLensEqual = (
           return isEqualFromVis(
             doc1.state.visualization,
             doc1.references,
+            doc1.state.datasourceStates?.formBased,
```

Comment:
why ignore the text base one?
```suggestion
            doc1.state.datasourceStates,
```

Reply:
To my knowledge the `colorMapping` feature does not support `textBased`. If you think it's better to just pass all the states here and use the `formBased` downstream I'm fine with that too.

Reply:
I do not think it would change much to be honest.
Also datasources cannot be mixed, so either is `formBased` or `textBased`, so not a big deal I guess.
In the long run `formBased` and `textBased` will be merged.

Reply:
Discussed this offline and opened #218660 to improve consistency of the `datasourceState` types across Lens.

Reply:
This was not true, you can set color mappings in ES|QL vis on a dashboard. I refactored the runtime converter logic to use either `textBased` or `formBased`. This is needed to backout the original raw value from it's string form. The `textBased` usage does not require knowing if the column is a `range` or `multi-field` value.

See changes in 878a48556a848b8d34945cb7d90713e1a254928b